src/protocol/id/contract.md

```diff
@@ -1,3 +1,3 @@
 # Contract ID
 
-For a transaction of type `TransactionType.Create`, `tx`, the contract ID is `sha256(0x4655454C ++ tx.data.salt ++ root(tx.data.witnesses[bytecodeWitnessIndex].data) ++ root_smt(tx.storageSlots))`, where `root` is the Merkle root of [the binary Merkle tree](../cryptographic_primitives.md#binary-merkle-tree) with each leaf being an 8-byte word of two instructions, and `root_smt` is the [Sparse Merkle tree](../cryptographic_primitives.md#sparse-merkle-tree) root of the provided key-value pairs. If the bytecode is not a multiple of 8 bytes (i.e. if there are an odd number of instructions), the last instruction is padded with 4-byte zero.
+For a transaction of type `TransactionType.Create`, `tx`, the contract ID is `sha256(0x4655454C ++ CHAIN_ID ++ tx.data.salt ++ root(tx.data.witnesses[bytecodeWitnessIndex].data) ++ root_smt(tx.storageSlots))`, where `root` is the Merkle root of [the binary Merkle tree](../cryptographic_primitives.md#binary-merkle-tree) with each leaf being an 8-byte word of two instructions, and `root_smt` is the [Sparse Merkle tree](../cryptographic_primitives.md#sparse-merkle-tree) root of the provided key-value pairs. If the bytecode is not a multiple of 8 bytes (i.e. if there are an odd number of instructions), the last instruction is padded with 4-byte zero.
```

Comment:
What are the benefits of adding chain id into the `ContractId` calculation? And what vulnerabilities do we have if we don't do that?

With the same `ContractId` is simpler for developers to deploy the several instances of the same project on different networks.

Reply:
I see that it is important for the predicate to be unique per network because it allows using inputs. But I'm not sure that we need this rule for contracts.

Reply:
True, we don't need chain specific contract ids to prevent replay attacks. The only benefit would be to ensure there is a globally unique id for contracts which could help with search & indexing of multiple chains.

Reply:
> The only benefit would be to ensure there is a globally unique id for contracts which could help with search & indexing of multiple chains.

Would this really be useful? If you're searching for a contract, you're probably searching for it on a particular chain. If anything, perhaps it could be useful for the contract ID to be universal, as you could more easily identify instances of the same contract on different networks?

It would be nice to have a good motivation to change the behaviour of contract IDs, as it does complicate some tooling features around `CONTRACT_ID` consts a little. That said, I guess we'll need to make a bunch of the same changes for predicate roots either way.